### DISCUSSION TITLE
Suche Code zum selektiven Auslesen der Solarplatten beim Hoymiles HMT 2250T.

### DISCUSSION THREAD

**Discussion by Zurrmaxe** [hoylabs/OpenDTU-OnBattery#1098] [Q&A]
Hallo zusammen

Für eine externe Anzeige möchte ich gerne die Solarplatten am HMT2250T über http.getString  /JSON pp. einzeln auslesen.

Dazu habe ich schon verschiedene Versuche gemacht:

z.B.
       platte 1 = doc["inverters"]["Power"];
       platte 1 = doc["inverters"]["power"]["v"];
       platte 1 = doc[138290780486]['1']['Power']['v'];
       usw...

Alles hat nicht geklappt. Es wird nichts gelesen. 
Da ist irgendwo der Wurm drin. 

Hat einer die Idee zum richtigen Code ?

Hinweis:  Diese Codes

        atpwr =doc["inverters"][0]["AC"]["0"]["Power"]["v"]; // POWER
        adpwr =doc["total"]["YieldDay"]["v"]; //TAGESLEISTUNG
        atemp =doc["inverters"][0]["INV"]["0"]["Temperature"]["v"]; //Inverter Temperatur

funktionieren und können problemlos ausgelesen werden.

mfg.
Alfred

**Comment by Tiese** [comment]
Hallo @Zurrmaxe,

schau doch mal, was unter: `http://dieIPvonOpdenDTUoB/api/livedata/status?inv=WR-Ser.-Nr.` ausgegeben wird.

**Reply by Zurrmaxe** [reply]
Besten Dank für die Info. 
Habe den Pfad gefunden und es klappt, zumindest in technischer Hinsicht. 
Jetzt werden mit in meiner Balkenanzeige auch die jeweiligen Platten 1 -12 angezeigt.
Allerdings sind durch die ständige Regelungen der DTU (Nulleinspeisung.... ) die ganzen Balken immer in Bewegung.  
Sehr unschön für die Optik. Mal schauen was sich da machen lässt.
mfg. Alfred

**Reply by spcqike** [reply]
12 Platten? Hat der HMT-2250 nicht nur 6 Eingänge? 

> Allerdings sind durch die ständige Regelungen der DTU (Nulleinspeisung.... ) die ganzen Balken immer in Bewegung.

das ist halt so. Willst du die echten live Daten? Dann zappeln sie mit deiner Anforderung / Verbrauch / Erzeugung. 
willst du schöne glatte Anzeigen? Dann musst du die Regelung träger machen oder die Auswertung mitteln / glätten.

**Reply by Zurrmaxe** [reply]
Die Livedaten sind in der Tat nicht aussagekräftig.  Zuviel gezappeI.
Daher habe ich mich daher für die "Tagesleistung" der Platten entschieden. 
Diese sind auch auf längerer Sicht besser zu vergleichen.

Diese Anfrage kann von mir aus geschlossen werden . Danke für die Hilfen.
lg Alfred

**Reply by spcqike** [reply]
Speicher denn was dagegen einfach die gesamt Erträge fortlaufend zu protokollieren und den Rest der Auswertung in der Auswertung selbst zu machen? 

Der Tagesertrag ist auch nur der aktuelle Wert minus dem Wert von heute 00:00 Uhr. 

Der Wochenertrag ist dann der letzte minus dem von Montag 00:00 Uhr 

Und so weiter 

so zumindest mache ich es immer und überall. Das geht auch einfacher als dann bspw den Monatserttag als Summe aller einzelnen tageserträge zu berechnen.
